src/rust/ide/view/graph-editor/src/component/breadcrumbs/project_name.rs

```diff
@@ -305,91 +219,144 @@ impl display::Object for ProjectNameModel {
+                text.set_focus(false);
+                text.remove_all_cursors();
+                animations.color.set_target_value(deselected_color);
             });
-            eval_ frp.inputs.cancel_editing(model.reset_name());
-            eval  frp.inputs.name((name) {model.rename(name)});
-            eval_ frp.inputs.commit(model.commit());
-            frp.outputs.mouse_down <+ model.view.events.mouse_down;
-            eval_ model.view.events.mouse_down(frp.outputs.edit_mode.emit(true));
-            outside_press     <- any(&frp.outside_press,&frp.deselect);
-            commit_if_changed <- outside_press.gate(&frp.outputs.edit_mode);
-            eval_ commit_if_changed(frp.commit.emit(()));
-        }
+            frp.output.source.selected <+ deselect.to_false();
 
 
-        // === Animations ===
+            // === Animations ===
+
+            eval animations.color.value((value) model.set_color(*value));
+            eval animations.position.value((value) model.set_position(*value));
 
-        frp::extend! {network
-            eval model.animations.color.value((value) model.set_color(*value));
-            eval model.animations.position.value((value) model.set_position(*value));
         }
 
-        Self{frp,model}.init()
-    }
+        frp.deselect();
 
-    fn init(self) -> Self {
-        let project_name = Rc::downgrade(&self.model);
-        //FIXME[dg]: This section to check newline and keep TextField in a single line is hacky
-        // and should be removed once the new TextField is implemented.
-        self.text_field.set_text_edit_callback(move |change| {
-            if let Some(project_name) = project_name.upgrade() {
-                // If the text edit callback is called, the TextEdit must be still alive.
-                let field_content = project_name.text_field.get_content();
-                let new_name      = field_content.replace("\n", "");
-                // Keep only one line.
-                project_name.text_field.set_content(&new_name);
-                project_name.outputs.width.emit(project_name.width());
-                project_name.update_alignment();
-                if change.inserted == "\n" {
-                    project_name.commit();
-                }
-            }
-        });
-        self
+        Self{frp,model}
     }
+
 }
 
 impl display::Object for ProjectName {
     fn display_object(&self) -> &display::object::Instance {
-        &self.display_object
+        &self.model.display_object
+    }
+}
+
+impl Deref for ProjectName {
+    type Target = Frp;
+    fn deref(&self) -> &Self::Target {
+        &self.frp
+    }
+}
+
+impl application::command::FrpNetworkProvider for ProjectName {
+    fn network(&self) -> &frp::Network { &self.frp.network }
+}
+
+impl View for ProjectName {
+    fn label() -> &'static str { "ListView" }
+
+    fn new(app:&Application) -> Self { ProjectName::new(app) }
+
+    fn app(&self) -> &Application { &self.model.app }
+
```

Comment:
if we use such one-liners, we recently glue them together and we align -> together - it looks really nice! :)